src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py

```diff
@@ -0,0 +1,718 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module provides the intermediate representations and analysis functions for GitHub Actions."""
+
+import logging
+import os
+import re
+from collections.abc import Iterable
+from dataclasses import dataclass
+from enum import Enum
+
+from macaron.code_analyzer.call_graph import BaseNode
+from macaron.config.global_config import global_config
+from macaron.errors import CallGraphError, GitHubActionsValueError, ParseError
+from macaron.parsers.actionparser import get_step_input
+from macaron.parsers.actionparser import parse as parse_action
+from macaron.parsers.bashparser import BashNode, BashScriptType, create_bash_node
+from macaron.slsa_analyzer.build_tool.language import BuildLanguage, Language
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ThirdPartyAction:
+    """The representation for a third-party GitHub Action."""
+
+    #: The name of the GitHub Action.
+    action_name: str
+
+    #: The version of the GitHub Action.
+    action_version: str | None
+
+
+class GHWorkflowType(Enum):
```

Comment:
nit: we use a short form of GitHub (`GH`) here while compared to `GitHubWorkflowNode`. It could be renamed for consistency (in my opinion).

Reply:
See commit [f0d4b0e](https://github.com/oracle/macaron/pull/673/commits/f0d4b0e3f9afb13e908daceccef9c387d7423fd4)